### PR TITLE
Adds new direct mob sound admin verb

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Adds the "Play Direct Mob Sound" for (b)admins to enjoy. Pick a sound and a mob, and only the chosen mob will hear it. Button also available in the player panel.

## Why It's Good For The Game
Happy admins happy life

## Changelog
:cl: Skoglol
admin: New sound verb, Play Direct Mob Sound. Can also be found in the player panel.
/:cl:

### CODE REVIEW

**Comment by Tharcoonvagh** [comment]
Based

**Comment by Dorsisdwarf** [comment]
Shouldnt this be "Play Direct Mob Sound"? Play local sound is already a button, so this could be confusing in the future.

**Comment by kriskog** [comment]
Honestly trying to properly describe all these in one or two words is a lost cause. Should probably just combine them all in one verb and make it a selection.
I went with local mob sound since local is how it is referred to in code, but I'm not dead set on that. It doesnt roll off the tongue very easily

**Comment by zxaber** [comment]
Could this be a function in the client's player panel? Seems like that would be more straight-foward, especially if you're already doing memes with this particular player.

**Comment by kriskog** [comment]
Good idea, Ill look into it

File: code/modules/admin/verbs/playsound.dm [hunk 1]
```diff
@@ -52,6 +52,22 @@
...
+		return
+
+	var/mob/target = input(usr, "Choose a mob to play the sound to.", "Play Mob Sound") as null|anything in sortNames(GLOB.player_list)
+	var/vol = input(usr, "What volume would you like the sound to play at?",, 50) as null|num
+	if(!vol)
+		return
+	vol = CLAMP(vol, 1, 100)
+	log_admin("[key_name(src)] played a local mob sound [S] to [target].")
+	message_admins("[key_name_admin(src)] played a local mob sound [S] to [ADMIN_LOOKUPFLW(target)].")
+	target.playsound_local(get_turf(target), S, vol, FALSE, pressure_affected = FALSE)
```
Comment (ninjanomnom) [inline code/modules/admin/verbs/playsound.dm]: You want `SEND_SOUND()` instead

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/admin/verbs/playsound.dm (hunk 1 above)
Comment (kriskog) [reply code/modules/admin/verbs/playsound.dm]: done

**Comment by kriskog** [comment]
> 
> 
> Could this be a function in the client's player panel? Seems like that would be more straight-foward, especially if you're already doing memes with this particular player.

Now also in player panel.

> Shouldnt this be "Play Direct Mob Sound"? Play local sound is already a button, so this could be confusing in the future.

Now called direct mob sound, thanks

File: code/modules/admin/verbs/playsound.dm [hunk 2]
```diff
@@ -52,6 +52,19 @@
...
 
+/client/proc/play_direct_mob_sound(S as sound, mob/M)
+	set category = "Fun"
+	set name = "Play Direct Mob Sound"
+	if(!check_rights(R_SOUND))
+		return
+
+	if(!M)
+		M = input(usr, "Choose a mob to play the sound to. Only they will hear it.", "Play Mob Sound") as null|anything in sortNames(GLOB.player_list)
+	log_admin("[key_name(src)] played a direct mob sound [S] to [M].")
```
Comment (Cyberboss) [inline code/modules/admin/verbs/playsound.dm]: Check if M isn't null before proceeding.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/admin/verbs/playsound.dm (hunk 2 above)
Comment (kriskog) [reply code/modules/admin/verbs/playsound.dm]: Is that fine?

**Pull request merged**